src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -186,10 +186,9 @@ class NBDServer
 
     if (ret < 0) {
       ctx->reply.error = htonl(-ret);
-    } else if ((ctx->command == NBD_CMD_WRITE || ctx->command == NBD_CMD_READ)
-	       && ret != static_cast<int>(ctx->request.len)) {
+    } else if ((ctx->command == NBD_CMD_WRITE) && ret != 0) {
```

Comment:
@cy-lee shouldn't this be something like below instead?
```
else if (ctx->command == NBD_CMD_READ && ret  != ctx->request.len) 
  error "unexpected return value for read"
else if (ctx->command != NBD_CMD_READ && ret  != 0)
  error "unexpected return value"
```
I.e. test that read returns number of bytes requested, any other ops should return 0 on success.

Or may be just check that return value 0 or request.len would be enough?